### ISSUE TITLE
Local Peaks tool contains duplicative points.

### ISSUE THREAD

**Issue by NatalieCampos** [Esri/military-tools-geoprocessing-toolbox#221]
Running the Local Peaks tool in ArcMap, it seems that the results contain points for peaks that are close together and of similar value. This gives the impression that a single peak is counted twice and results in erroneous results as the number of peaks and location of the peaks is wrong if peaks are counted multiple times. 

## Expected Behavior
I would expect a single point per mountain peak.

## Current Behavior
Multipel points per mountain peak.

Example 1: Please notice location and elevation values.  
![image](https://cloud.githubusercontent.com/assets/5116835/26226763/043ac21a-3be3-11e7-8855-9ab10fbeb676.png)

Example 2:  Please notice location and elevation values.  

![image](https://cloud.githubusercontent.com/assets/5116835/26226779/1d773efc-3be3-11e7-9b0c-6f57dffdb148.png)

Example 3:  Please notice location and elevation values.  
![image](https://cloud.githubusercontent.com/assets/5116835/26226798/3b9fa75c-3be3-11e7-832b-c4b5e91a959f.png)

## Context
This issue is a result of the data provided by @thumber1776

## Your Environment
ArcMap 10.3.1

**Comment by adgiles** [comment]
@NatalieCampos @nfeuerstein 

Natalie(s). The find highest peaks tool has a number of flaws in its logic. The current workflow is to:

- invert the elevation model
- run the Flow Direction to enable sink areas to be determined
- create points from the sink areas to determine the original elevation values
- then select all the elevation values with the top (x) values (x being the number of peaks the user wants)

The issues this workflow has is that if you have adjacent raster cells forming one sink area each of the individual pixels is going to be turned into a individual peak (hence this issue and #219).

The next problem is then if you select all the peaks that match the top x values the result could be higher than x (if you have multiple peaks of the same value). Instead you only want to select the top x values (second part of issue #217).

So I think the solution is to turn the sink areas to a polygon feature class then create a point feature class from this, as this will ensure that you only get a single central point if the single area is over a number of pixels.

Once you are at the stage of selecting the number of required points, do not select points that have values equal to the highest points but only select the top amount required


I have made the changes and should be good to verify once PR #222 has been merged
